### PR TITLE
Implemented aws/aws-logging-dotnet#44

### PR DESCRIPTION
**Pull request by Herdo**
*Issue #44 :

*Description of changes:*
- Implemented logic to support ILogger.BeginScope<T>
- Extended configuration with "IncludeScopes" property, as well as
reading the new configuration property
- Implemented test cases
- Some minor code clean-up in the configuration classes (getter and setter for readability) & tests (removed unused namespaces)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/AWS.Logger.Core/AWSLoggerConfig.cs [hunk 1]
```diff
@@ -93,6 +93,33 @@ public class AWSLoggerConfig : IAWSLoggerConfig
...
+        /// </summary>
+        public string LibraryLogFileName { get; set; } = "aws-logger-errors.txt";
+
+        /// <summary>
+        /// Gets the <see cref="IAWSLoggerConfig.IncludeScopes"/> property. This determines if scopes - if they exist - are included in a log message.
+        /// <para>
+        /// The default is false.
+        /// </para>
+        /// </summary>
+        public bool IncludeScopes { get; set; } = false;
```
Comment (normj) [inline src/AWS.Logger.Core/AWSLoggerConfig.cs]: Why is this added to Core package? It doesn't seem like it is used at all.

**Comment by normj** [comment]
I have merged the PR into dev. I did rework the change a bit to keep all of the IncludeScope changes to be in just the ASP.NET Core logger.

**Comment by snakefoot** [comment]
NetCore 2.1 introduces a new interface `ISupportExternalScope` that improves the BeginScope-performance.

Example from EventLogLoggerProvider:

https://github.com/aspnet/Extensions/blob/master/src/Logging/Logging.EventLog/src/EventLogLoggerProvider.cs

https://github.com/aspnet/Extensions/blob/a2b9e4fb906eced0c602036897887c4c879f798d/src/Logging/Logging.EventLog/src/EventLogLogger.cs#L112

**Comment by snakefoot** [comment]
@normj Maybe consider changing the visibility:

`public class AWSLogScope`

To:

`internal sealed class AWSLogScope`

Since no one should use it directly.

Also consider fixing `DisposableScope` so one cannot perform double dispose.

**Comment by normj** [comment]
Good call @snakefoot  https://github.com/aws/aws-logging-dotnet/commit/5d29b661b4acc4048fc9b9bafdfa92f9c0603c89

**Comment by normj** [comment]
This was released today with version 1.4.0 of AWS.Logger.AspNetCore package. Thanks for the pull request.
